coreneuron/nrniv/nrn_acc_manager.cpp

```diff
@@ -981,4 +954,73 @@ void nrn_sparseobj_copyto_device(SparseObj* so) {
     }
 #endif
 }
+
+#ifdef _OPENACC
+
+void nrn_ion_global_map_copyto_device() {
+    if (nrn_ion_global_map_size) {
+        double** d_data =
+            (double**)acc_copyin(nrn_ion_global_map, sizeof(double*) * nrn_ion_global_map_size);
+        for (int j = 0; j < nrn_ion_global_map_size; j++) {
+            if (nrn_ion_global_map[j]) {
+                /* @todo: fix this constant size 3 :( */
+                double* d_mechmap = (double*)acc_copyin(nrn_ion_global_map[j], 3 * sizeof(double));
+                acc_memcpy_to_device(&(d_data[j]), &d_mechmap, sizeof(double*));
+            }
+        }
+    }
+}
+
+void init_gpu(int nthreads, NrnThread* threads) {
+    if (nthreads <= 0) {
+        printf("\n Warning: No threads to copy on GPU! ");
+        return;
+    }
+
+    /** @todo: currently only checking nvidia gpu */
+    int num_gpus = acc_get_num_devices(acc_device_nvidia);
+    if (num_gpus == 0) {
+        printf("\n WARNING: Enabled GPU execution but couldn't find NVIDIA GPU! \n");
+    }
+
+    /* @todo: why dt is not setup at this moment? */
+    for (int i = 0; i < nthreads; i++) {
+#ifndef UNIFIED_MEMORY
+        (threads + i)->_dt = dt;
+#endif
```

Comment:
can you check if this required? i.e. in unified and non-unified, could you check if dt == (threads+i)->dt  OR not?